2025-01-31 09:40:54,183 - ERROR - Exception on / [POST]
Traceback (most recent call last):
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask_wtf\csrf.py", line 108, in validate_csrf
    token = s.loads(data, max_age=time_limit)
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\itsdangerous\timed.py", line 204, in loads
    base64d, timestamp = signer.unsign(
                         ~~~~~~~~~~~~~^
        s, max_age=max_age, return_timestamp=True
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\itsdangerous\timed.py", line 142, in unsign
    raise SignatureExpired(
    ...<3 lines>...
    )
itsdangerous.exc.SignatureExpired: Signature age 3684 > 3600 seconds

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask_wtf\csrf.py", line 261, in protect
    validate_csrf(self._get_csrf_token())
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask_wtf\csrf.py", line 110, in validate_csrf
    raise ValidationError("The CSRF token has expired.") from e
wtforms.validators.ValidationError: The CSRF token has expired.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 865, in full_dispatch_request
    rv = self.preprocess_request()
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 1239, in preprocess_request
    rv = self.ensure_sync(before_func)()
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask_wtf\csrf.py", line 229, in csrf_protect
    self.protect()
    ~~~~~~~~~~~~^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask_wtf\csrf.py", line 264, in protect
    self._error_response(e.args[0])
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask_wtf\csrf.py", line 307, in _error_response
    raise CSRFError(reason)
flask_wtf.csrf.CSRFError: 400 Bad Request: The CSRF token has expired.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 752, in handle_user_exception
    return self.handle_http_exception(e)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 727, in handle_http_exception
    return self.ensure_sync(handler)(e)
           ~~~~~~~~~~~~~~~~~~~~~~~~~^^^
  File "C:\Users\pwwpp\Websites\anmeldung-kinderfasching\app.py", line 244, in handle_csrf_error
    return render_template("form.html"), 400
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\pwwpp\Websites\anmeldung-kinderfasching\templates\form.html", line 20, in top-level template code
    {{ form.hidden_tag() }}
    ^^^^^^^^^^^^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\jinja2\environment.py", line 490, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2025-01-31 09:40:54,246 - ERROR - 500 Error: http://127.0.0.1:10000/
2025-01-31 11:43:43,937 - ERROR - Exception on / [GET]
Traceback (most recent call last):
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 870, in full_dispatch_request
    return self.finalize_request(rv)
           ~~~~~~~~~~~~~~~~~~~~~^^^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 889, in finalize_request
    response = self.make_response(rv)
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 1161, in make_response
    raise TypeError(
    ...<3 lines>...
    )
TypeError: The view function for 'register' did not return a valid response. The function either returned None or ended without a return statement.
2025-01-31 11:43:43,986 - ERROR - 500 Error: http://127.0.0.1:10000/
2025-01-31 11:45:03,954 - ERROR - Exception on / [GET]
Traceback (most recent call last):
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 870, in full_dispatch_request
    return self.finalize_request(rv)
           ~~~~~~~~~~~~~~~~~~~~~^^^^
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 889, in finalize_request
    response = self.make_response(rv)
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 1161, in make_response
    raise TypeError(
    ...<3 lines>...
    )
TypeError: The view function for 'register' did not return a valid response. The function either returned None or ended without a return statement.
2025-01-31 11:45:03,957 - ERROR - 500 Error: http://127.0.0.1:10000/
2025-01-31 12:00:04,268 - ERROR - Exception on / [POST]
Traceback (most recent call last):
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\pwwpp\AppData\Local\Programs\Python\Python313\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\pwwpp\Websites\anmeldung-kinderfasching\app.py", line 193, in register
    return jsonify(success=False, errors=errors), 400
           ^^^^^^^
NameError: name 'jsonify' is not defined
2025-01-31 12:00:04,282 - ERROR - 500 Error: http://127.0.0.1:10000/
2025-01-31 13:04:58,423 - ERROR - Formularfehler: {'child_firstname': ['This field is required.'], 'child_lastname': ['This field is required.'], 'birthdate': ['This field is required.'], 'allergies': [], 'club_membership': ['This field is required.'], 'parent_firstname': [], 'parent_lastname': [], 'phone_number': [], 'email': [], 'csrf_token': []}
2025-01-31 13:05:36,187 - ERROR - Formularfehler: {'child_firstname': ['This field is required.'], 'child_lastname': ['This field is required.'], 'birthdate': ['This field is required.'], 'allergies': [], 'club_membership': ['This field is required.'], 'parent_firstname': [], 'parent_lastname': [], 'phone_number': [], 'email': [], 'csrf_token': []}
2025-01-31 13:11:26,626 - ERROR - Formularfehler: {'child_firstname': ['This field is required.'], 'child_lastname': ['This field is required.'], 'birthdate': ['This field is required.'], 'allergies': [], 'club_membership': ['This field is required.'], 'parent_firstname': [], 'parent_lastname': [], 'phone_number': [], 'email': [], 'csrf_token': []}
2025-01-31 13:17:00,665 - ERROR - Formularfehler: {'child_firstname': ['This field is required.'], 'child_lastname': ['This field is required.'], 'birthdate': ['This field is required.'], 'allergies': [], 'club_membership': ['This field is required.'], 'parent_firstname': [], 'parent_lastname': [], 'phone_number': [], 'email': [], 'csrf_token': []}
2025-01-31 13:19:52,514 - ERROR - Formularfehler: {'child_firstname': ['This field is required.'], 'child_lastname': ['This field is required.'], 'birthdate': ['This field is required.'], 'allergies': [], 'club_membership': ['This field is required.'], 'parent_firstname': [], 'parent_lastname': [], 'phone_number': [], 'email': [], 'csrf_token': []}
